src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,343 @@
+                            await client.Issue.Comment.Create(repo.Id, issueNumber.Value, description.ToString());
+                        }
+                    }
+                }
+                else
+                {
+                    // Create a new issue for the error
+                    string labelName = "DependencyUpdateError"; 
+                    Logger.LogInformation("Dependency Update Error, for the error message " + repositoryBranchUpdateHistory.ErrorMessage +
+                        " for the repository : " + repositoryBranchUpdateHistory.Repository);
+                    var createIssue = new NewIssue("[Dependency Update] Errors during dependency updates to :" + repositoryBranchUpdateHistory.Repository);
+                    string body = "The following errors have been detected when attempting to update dependencies in " +
+                        repositoryBranchUpdateHistory.Repository + "\n\n";
+                    createIssue.Body = body + description.ToString() + "\n\n**/FyiHandle :** " + _options.FyiHandle;
+                    createIssue.Labels.Add(labelName);
+                    var issue = await client.Issue.Create(repo.Id, createIssue);
+                    Logger.LogInformation("Issue Number " + issue.Number + " was created in " + issueRepo);
+                    await gitHubIssueEvaluator.SetAsync(tx, (repositoryBranchUpdateHistory.Repository, repositoryBranchUpdateHistory.Branch), issue.Number);
+                }
+                await tx.CommitAsync();
+            }
+        }
+
+        private bool UpdateIssueCommentBody(RepositoryBranchUpdateHistoryEntry repositoryBranchUpdateHistory, string body)
+        {
+            Dictionary<string, string> argumentParser = parseIssueCommentBody(body);
+            // Check if the error for the method UpdateAssetsAsync exists 
+            if (string.Equals(argumentParser["method"], "UpdateAssetsAsync"))
+            {
+                string subId = GetSubscriptionId(repositoryBranchUpdateHistory.Arguments);
+                if (string.Equals(argumentParser["subscriptionId"], subId) &&
+                    string.Equals(argumentParser["errorMessage"], repositoryBranchUpdateHistory.ErrorMessage))
+                {
+                    return true;
+                }
+            }
+            // Check if the error for the method ProcessPendingUpdatesAsync exists 
+            if (string.Equals(argumentParser["method"], "ProcessPendingUpdatesAsync") &&
+                string.Equals(argumentParser["errorMessage"], repositoryBranchUpdateHistory.ErrorMessage))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private Dictionary<string, string> parseIssueCommentBody (string body)
+        {
+            Dictionary<string, string> arguments = new Dictionary<string, string>();
+            Regex rx = new Regex(@"(\w+): ([^,]+)(?:, |\))",
```

Comment:
If you are going to use a regex, this needs a comment explaining what its looking for, including some examples.

Reply:
Also, there's no reason why the pattern should be in the method, you can also probably specialize the pattern a bit, but it's hard to know without having an example of what you're trying to match.

---
In reply to: [391962441](https://github.com/dotnet/arcade-services/pull/1041#discussion_r391962441) [](ancestors = 391962441)

Reply:
Added an example.